apps/bible/models.py

```diff
@@ -238,6 +238,20 @@ def from_reference(cls, reference, persist=False):
             verse_end_separator = match.group('verse_end_separator')
             verse_end = match.group('verse_end')
             verse_end = int(verse_end) if verse_end else None
+            has_ff = bool(match.group('ff'))
+            
+            if has_ff:
+                # resolve the passage recursively
+                non_abbreviated_passage = cls.from_reference(reference.removesuffix('ff'), persist=False)
+                if non_abbreviated_passage.chapter_end:
+                    chapter_end = non_abbreviated_passage.chapter_end.chapter
+                    verse_end = non_abbreviated_passage.chapter_end.bibleverses.last().verse
+                else:
+                    chapter_end = non_abbreviated_passage.chapter_start.chapter
+                    verse_end = non_abbreviated_passage.chapter_start.bibleverses.last().verse
```

Comment:
```suggestion
                chapter_end_obj = non_abbreviated_passage.chapter_end or non_abbreviated_passage.chapter_start
                chapter_end = chapter_end_obj.chapter
                verse_end = chapter_end_obj.bibleverses.last().verse
```

Further simplify. If you are only doing assignment, use the "ternary" expression or `or`. Put all of the switch/decision code inside of the "parenthetical phrase," and all common code outside of that.